openslides_backend/action/actions/motion/create_base.py

```diff
@@ -0,0 +1,105 @@
+    SequentialNumbersMixin,
+    SetNumberMixin,
+    CreateActionWithListOfSpeakersMixin,
+):
+
+    model = Motion()
+    dependencies = [AgendaItemCreate, ListOfSpeakersCreate]
+
+    def set_state_from_workflow(
+        self, instance: Dict[str, Any], meeting: Dict[str, Any]
+    ) -> None:
+        workflow_id = instance.pop("workflow_id", None)
+        if workflow_id is None:
+            if instance.get("lead_motion_id"):
+                workflow_id = meeting.get("motions_default_amendment_workflow_id")
+            elif instance.get("statute_paragraph_id"):
+                workflow_id = meeting.get(
+                    "motions_default_statute_amendment_workflow_id"
+                )
+            else:
+                workflow_id = meeting.get("motions_default_workflow_id")
+        if workflow_id:
+            workflow = self.datastore.get(
+                FullQualifiedId(Collection("motion_workflow"), workflow_id),
+                ["first_state_id"],
+            )
+            instance["state_id"] = workflow.get("first_state_id")
+        else:
+            raise ActionException(
+                "No matching default workflow defined on this meeting"
+            )
+
+    def check_for_origin_id(self, instance: Dict[str, Any]) -> None:
+        if instance.get("origin_id"):
+            meeting = self.datastore.get(
+                FullQualifiedId(Collection("meeting"), instance["meeting_id"]),
+                ["committee_id"],
+            )
+            forwarded_from = self.datastore.get(
+                FullQualifiedId(Collection("motion"), instance["origin_id"]),
+                ["meeting_id"],
+            )
+            forwarded_from_meeting = self.datastore.get(
+                FullQualifiedId(Collection("meeting"), forwarded_from["meeting_id"]),
+                ["committee_id"],
+            )
+            committee = self.datastore.get(
+                FullQualifiedId(
+                    Collection("committee"), forwarded_from_meeting["committee_id"]
+                ),
+                ["forward_to_committee_ids"],
+            )
+            if meeting["committee_id"] not in committee.get(
+                "forward_to_committee_ids", []
+            ):
+                raise ActionException(
+                    f"Committee id {meeting['committee_id']} not in {committee.get('forward_to_committee_ids', [])}"
+                )
+
+    def create_submitters(self, instance: Dict[str, Any]) -> None:
+        submitter_ids = instance.pop("submitter_ids", None)
+        if not submitter_ids:
+            submitter_ids = [self.user_id]
+        self.apply_instance(instance)
+        action_data = []
+        for user_id in submitter_ids:
+            action_data.append({"motion_id": instance["id"], "user_id": user_id})
+        self.execute_other_action(MotionSubmitterCreateAction, action_data)
+
+        instance["sequential_number"] = self.get_sequential_number(
+            instance["meeting_id"]
+        )
```

Comment:
This does not belong in this function. You can probably add it to `set_created_last_modified_and_number` and rename that to something like `set_additional_motion_fields` (and add a docstring explaining which fields are set)

Reply:
Add set_sequential_number() function and set this number there.